q2_mystery_stew/plugin_setup.py

```diff
@@ -451,6 +454,30 @@ def register_single_type_tests(plugin, list_of_params):
             )
 
 
+def register_typemap_tests(plugin):
+    input_map, output_map = \
+        TypeMap({SingleInt1: EchoOutput,
+                 List[SingleInt1 | SingleInt2] | List[SingleInt1]: EchoOutput})
```

Comment:
I believe this expression collapses into `List[SingleInt1 | SingleInt2]` all on its own. You would need a `List[SingleInt3]` for it to not be declared redundant by the outer union here. (It's ok though, we don't really need that second half of the outer union.)